zerver/lib/webhooks/common.py

```diff
@@ -75,10 +77,27 @@ def check_send_webhook_message(
     user_profile: UserProfile,
     topic: str,
     body: str,
+    complete_event_type: Optional[str] = None,
     stream: Optional[str] = REQ(default=None),
     user_specified_topic: Optional[str] = REQ("topic", default=None),
+    only_events: Optional[List[str]] = REQ(default=None, json_validator=check_list(check_string)),
+    exclude_events: Optional[List[str]] = REQ(
+        default=None, json_validator=check_list(check_string)
+    ),
     unquote_url_parameters: bool = False,
 ) -> None:
+    if complete_event_type is not None:
+        # If complete_event_type is passed to this function, we will check the event type against user configured lists of only_events
+        # and exclude events. If the event does not satisfy the configuration, the function will return without doing anything.
+        # We consider items in only_events and exclude_events to be regular expressions and forces full-text match for each of them.
+        if (
+            only_events is not None
+            and all([not re.match(f"^{event}$", complete_event_type) for event in only_events])
+        ) or (
+            exclude_events is not None
+            and any([re.match(f"^{event}$", complete_event_type) for event in exclude_events])
+        ):
```

Comment:
Why do you need regular expressions here?  I would have thought it'd just be checks like `complete_event_type in exclude_events`.

Reply:
Because we allow events in `exclude_events` to be regular expressions. Then events like `"pu"` will actually match the first two characters of `"push"`, which is not the desired behavior. So I force the regex to match the entire string.

Reply:
User-specified regular expressions can be dangerous, since they can lead to DoS vulnerabilities and the like.  We could consider a more limited glob-style matching (https://docs.python.org/3/library/fnmatch.html), which would let you do `pull_request/*`.